omnibus/package-scripts/angrychef/preinst

```diff
@@ -7,14 +7,30 @@
 #   this programming language.  do not touch.
 # - if you are under 40, get peer review from your elders.
 
+# This whole file exists because on some versions of RPM, during upgrades, the postrm script
+# runs AFTER the installation of the upgraded RPM. Oops. See https://github.com/chef/chef/pull/7964
+
 INSTALLER_DIR=/opt/chef
+
 if [ -e $INSTALLER_DIR ]; then
   echo "removing $INSTALLER_DIR..."
+  rm -rf $INSTALLER_DIR || true
+fi
+
+# Sometimes, on some platforms, we get an EBUSY error trying to delete this, which will cause issues
+# with the upgrades. So we move it so it won't interfere with the upgrade, then delete it, which will
+# cause the OS to free the space once it's no longer used (when the current chef run ends, or whatever
+# else is using those libs). See: https://github.com/chef/chef/pull/8046
+if [ -e $INSTALLER_DIR ]; then
+  echo "Some files failed to delete, moving them so we don't interfere with upgrades, and then removing them"
 
   # have to do this dance of moving /opt/chef to a tmp dir since files may be in use
   tmp_dir="/opt/.chef.$$"
+  if [ ! -z $CHEF_UPGRADE_TEMP_DIR ]; then
+    tmp_dir="$CHEF_UPGRADE_TEMP_DIR/chef.$$"
+  fi
```

Comment:
This feels dangerous to me. Moving across filesystems isn't atomic. Plus we could end up leaving files around in a place we have no way of ever finding again, because it could have been a var in just that shell. I suspect the customer who hit the LVM limits will no longer hit it because of the initial `rm -rf $INSTALLER_DIR || true` above, so this _shouldn't_ be needed.

Reply:
fair point, we can approach this incrementally. if, at a later time, we need to add the env-var fetch, there will need to be some path validation to secure the script.